shell/platform/fuchsia/flutter/accessibility_bridge.cc

```diff
@@ -309,15 +320,62 @@ void AccessibilityBridge::AddSemanticsNodeUpdate(
     PrintNodeSizeError(nodes.back().node_id());
   }
 
+  // Even if the SemanticsNodeUpdates does not contain the root node, we still
+  // need to generate an update for the root node in case the view pixel ratio
+  // has changed.
+  std::vector<fuchsia::accessibility::semantics::Node> root_update;
+  root_update.push_back(GetRootNodeUpdate());
+
   PruneUnreachableNodes();
   UpdateScreenRects();
 
-  tree_ptr_->UpdateSemanticNodes(std::move(nodes));
+  tree_ptr_->UpdateSemanticNodes(std::move(root_update));
```

Comment:
Why not just push the root node update (or, the synthetic node if we go that route) onto the updates rather than making two calls to UpdateSemanticNodes?

Reply:
I don't want to go synthetic node route because we will then need to do node id conversion because the fuchsia always assume the first node id = 0.

If we want to reuse the updates list, we will need to handle size limit. I figure it may be easier to do it this way? (although test may need to be changed.)

Reply:
0 should still be the root node, but we insert a child node right between the root node and the rest of the nodes coming in. Although I guess we'd have to come up with an ID the framework would not use in that case, which is unfortunate. It might be safe enough to use the maximum allowed ID value - especially since this is only expected to be a short term thing until the embedding stops inserting the transform.

Reply:
and other reason i dislike synthetic node is that it may receive focus when doing swipe left or right. and it may possibly receiving hit testing, unless we make it not a a11y element. I am not sure if there is a flag for that in fuchsia.

My current intention is still go with this approach, but I am open to discussion

Reply:
I think I agree that we should avoid a synthetic node, because we'd potentially be using an ID the framework wants to use.

But I still don't quite see why we need a separate update.

There's already logic that should be handling the size limit right?

Reply:
Yeah, just move this logic up a bit in the method so we can append it to our nodes before we start splitting them.